llama_stack/providers/remote/vector_io/pgvector/pgvector.py

```diff
@@ -70,8 +75,77 @@ def load_models(cur, cls):
+  - Eg. SQL query: SELECT document, embedding <-> %s::vector AS distance FROM table ORDER BY distance
+
+-Characteristics:
+  - Semantic understanding - finds documents similar in meaning even if they don't share keywords
+  - Works with high-dimensional vector embeddings (typically 768, 1024, or higher dimensions)
+  - Best for: Finding conceptually related content, handling synonyms, cross-language search
+
+2. Keyword Search
+- How it works:
+  - Uses PostgreSQL's full-text search capabilities with tsvector and ts_rank
+  - Converts text to searchable tokens using to_tsvector('english', text)
+  - Eg. SQL query: SELECT document, ts_rank(content_tsvector, plainto_tsquery('english', %s)) AS score
+
+- Characteristics:
+  - Lexical matching - finds exact keyword matches and variations
+  - Uses GIN (Generalized Inverted Index) for fast text search performance
+  - Scoring: Uses PostgreSQL's ts_rank function for relevance scoring
+  - Best for: Exact term matching, proper names, technical terms, Boolean-style queries
+
+3. Hybrid Search
+- How it works:
+  - Combines both vector and keyword search results
+  - Runs both searches independently, then merges results using configurable reranking
+
+- Two reranking strategies available:
+    - Reciprocal Rank Fusion (RRF) - (default: 60.0)
+    - Weighted Average - (default: 0.5)
+
+- Characteristics:
+  - Best of both worlds: semantic understanding + exact matching
+  - Documents appearing in both searches get boosted scores
+  - Configurable balance between semantic and lexical matching
+  - Best for: General-purpose search where you want both precision and recall
+
+4. Database Schema
+The PGVector implementation stores data optimized for all three search types:
+CREATE TABLE vector_store_xxx (
+    id TEXT PRIMARY KEY,
+    document JSONB,                    -- Original document
+    embedding vector(dimension),        -- For vector search
+    content_text TEXT,                 -- Raw text content
+    content_tsvector TSVECTOR          -- For keyword search
+);
+
+-- Indexes for performance
+CREATE INDEX content_gin_idx ON table USING GIN(content_tsvector);  -- Keyword search
+-- Vector index created automatically by pgvector
+"""
+
+
 class PGVectorIndex(EmbeddingIndex):
-    def __init__(self, vector_db: VectorDB, dimension: int, conn, kvstore: KVStore | None = None):
+    # reference: https://github.com/pgvector/pgvector?tab=readme-ov-file#querying
+    PGVECTOR_DISTANCE_METRIC_TO_SEARCH_OPERATOR: dict[str, str] = {
+        "L2": "<->",  # Euclidean distance
+        "L1": "<+>",  # Manhattan distance
+        "COSINE": "<=>",  # Cosine distance
+        "INNER_PRODUCT": "<#>",  # Inner product distance
+        "HAMMING": "<~>",  # Hamming distance
+        "JACCARD": "<%>",  # Jaccard distance
+    }
+
+    def __init__(
+        self, vector_db: VectorDB, dimension: int, conn, kvstore: KVStore | None = None, distance_metric: str = "L2"
```

Comment:
Distance_metric is used while executing the query, but defining here means that we are restricting a particular DB to use only single metric. This doesn't seem to be the right place imo. 

We can either just restrict using just COSINE as milvus does it - https://github.com/meta-llama/llama-stack/blob/342550c1e24cc3e354c4afde92940da28ace9a32/llama_stack/providers/remote/vector_io/milvus/milvus.py#L107, or probably introduce it as a parameter in RAGConfig (https://github.com/meta-llama/llama-stack/blob/342550c1e24cc3e354c4afde92940da28ace9a32/llama_stack/apis/tools/rag_tool.py#L151) which we can use in other DBs too.

cc: @franciscojavierarceo @ehhuang